lighting_reporting_vanguard_product/models/lighting_product.py

```diff
@@ -0,0 +1,17 @@
+# Copyright NuoBiT Solutions - Frank Cespedes <fcespedes@nuobit.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class LightingProduct(models.Model):
+    _inherit = "lighting.product"
+
+    def get_finish_display(self):
+        self.ensure_one()
+        finish = " / ".join(
+            filter(None, [self.finish_id.display_name, self.finish2_id.display_name])
+        )
+        if self.ral_id:
+            finish += f" ({self.ral_id.display_name})"
+        return finish
```

Comment:
If you need a non-embedded Python code, remember always to use the accumulative method of building strings. More clear and clean.

```suggestion
    def get_finish_display(self):
        self.ensure_one()
        res, res_l = None, []
        
        # add finish description
        finish_l = []
        if self.finish_id:
            finish_l.append(self.finish_id.display_name)
        if self.finish2_id:
            finish_l.append(self.finish2_id.display_name)
        if finish_l:
            res_l.append("/".join(finish_l))
        
        # add RAL description   
        if self.ral_id:
            res_l.append("(%s)" % self.ral_id.display_name)
           
        if res_l:
            res = ' '.join(res_l)
            
        return res
```